sdk/lib/atl/atlwin.h

```diff
@@ -587,20 +587,31 @@ class CWindow
     }
 
 #ifdef __ATLSTR_H__
-    UINT GetDlgItemText(int nID, CSimpleString& string)
+    UINT GetDlgItemText(int nID, CSimpleString& strText) const
     {
         HWND item = GetDlgItem(nID);
+        if (!item)
+        {
+            strText.Empty();
+            return 0;
+        }
+#if 1
+        return CWindow(item).GetWindowText(strText);
+#else
         int len = ::GetWindowTextLength(item);
-        len = GetDlgItemText(nID, string.GetBuffer(len+1), len+1);
-        string.ReleaseBuffer(len);
+        len = GetDlgItemText(nID, strText.GetBuffer(len+1), len+1);
+        strText.ReleaseBuffer(len);
         return len;
+#endif
     }
 #endif
 
     BOOL GetDlgItemText(int nID, BSTR& bstrText) const
     {
-        ATLASSERT(::IsWindow(m_hWnd));
-        return FALSE;//FIXME stub
+        HWND item = GetDlgItem(nID);
+        if (!item)
+            return FALSE;
+        return CWindow(item).GetWindowText(bstrText);
```

Comment:
Same technique here, to not copy-paste the already-existing implementation of `GetWindowText(BSTR& bstrText)` that is "rather long"...